types/src/transaction/script.rs

```diff
@@ -125,6 +145,82 @@ impl ScriptABI {
     }
 }
 
+impl ScriptFunctionABI {
+    pub fn new(
+        name: String,
+        module_name: ModuleId,
+        doc: String,
+        ty_args: Vec<TypeArgumentABI>,
+        args: Vec<ArgumentABI>,
+    ) -> Self {
+        Self {
+            name,
+            module_name,
+            doc,
+            ty_args,
+            args,
+        }
+    }
+
+    pub fn name(&self) -> &str {
```

Comment:
For the records, I wish we could unify the diem-types data-structures and the generated ones. (I.e. I wish we could make all fields public and stop writing and using getters.)

Reply:
+1, should just use pub fields, as data structure is the contract for this case.